extensions/redis-client/runtime/src/main/java/io/quarkus/redis/runtime/datasource/AbstractGeoCommands.java

```diff
@@ -297,12 +297,23 @@ List<GeoValue<V>> decodeAsListOfGeoValues(Response r, boolean withDistance, bool
                         OptionalDouble.empty()));
             } else {
                 // Distance and Hash
-                double dist = response.get(1).toDouble();
+                double dist = parseDouble(response.get(1));
                 long hash = response.get(2).toLong();
                 list.add(new GeoValue<>(member, OptionalDouble.of(dist), OptionalLong.of(hash),
                         OptionalDouble.empty(), OptionalDouble.empty()));
             }
         }
         return list;
     }
+
+    private static double parseDouble(Response response) {
+        double dist;
+        try {
+            dist = response.toDouble();
+        } catch (NumberFormatException e) {
+            String s = response.toString().replaceAll("[^a-zA-Z0-9\\.]", "");
```

Comment:
Are you sure that the additional characters are only noise and can't change the meaning of the string?

Reply:
yes, it's just noise.

Reply:
OK so my only concern then is that we should compile the pattern for `replaceAll` since this will be called very often.

Reply:
I agree with using a compiled pattern, but it should not be called that often. The Redis change is only for distance computation when asking for hashes (I don't know why they only did it here)). I'm also pretty sure it's a bug, and they will rollback or fix it at some point.